altapay.php

```diff
@@ -0,0 +1,2390 @@
+                    $termNature = 'CreditCard';
+                } elseif (in_array('Invoice', $terminalNature)) {
+                    $termNature = 'Invoice';
+                }
+                $terminalArray[$terminal->getTitle()] = [
+                    'id'     => $terminal->getTitle(),
+                    'name'   => $terminal->getTitle(),
+                    'nature' => $termNature
+                ];
+            } else {
+                $terminalArray[$terminal->getTitle()] = $terminal;
+            }
+        }
+
+        return $terminalArray;
+    }
+
+
+    /* ******************************** */
+
+    /**
+     * @return string
+     */
+    private function getAPIUsername()
+    {
+        return $this->username;
+    }
+
+    /* ******************************** */
+
+    /**
+     * @return string
+     */
+    private function getAPIPassword()
+    {
+        return $this->password;
+    }
+
+    /* ******************************** */
+
+    /**
+     * @return string
+     */
+    public function getAltapayUrl()
+    {
+        return $this->url;
+    } /* ******************************** */
+
+    /**
+     * Get field values for add/edit terminal form
+     *
+     * @return array<string, mixed>
+     */
+    public function getFormValues()
+    {
+        $data       = [];
+        $idTerminal = (int)Tools::getValue('id_terminal');
+        if ($idTerminal > 0) {
+            $data = new Terminal($idTerminal);
+        } else {
+            $def = Terminal::$definition;
+            foreach ($def['fields'] as $fieldName => $stuff) {
+                $data[$fieldName] = Tools::getValue($fieldName);
+            }
+        }
+
+        return $data;
+    }
+
+    /**
+     * Handle payment processing
+     * capture, refund, release
+     *
+     * @return void
+     */
+    private function processPaymentActions()
+    {
+        $paymentID         = (int)Tools::getValue('payment_id');
+        $action            = Tools::ucfirst(Tools::getValue('action'));
+        $goodWillRefund    = false;
+        $orderID           = Tools::getValue('ap_order_id');
+        $orderLines        = Tools::getValue('ap_order_qty');
+        $orderLineGiftWrap = Tools::getValue('ap_order_wrap');
+
+        header('Content-Type: application/json');
+        if (!(Tools::getValue('action') && Tools::getValue('payment_id'))) {
+            return;
+        }
+        // Merchant API
+        $api = new MerchantAPI();
+        try {
+            $api->init($this->getAltapayUrl(), $this->getAPIUsername(), $this->getAPIPassword());
+        } catch (Exception $e) {
+            saveLastErrorMessage($paymentID, $e->getMessage());
+            echo json_encode(
+                [
+                    'status'  => 'error',
+                    'message' => 'Connection error: ' . $e->getMessage()
+                ]
+            );
+            exit();
+        }
+        if ($action == 'Capture') { // CAPTURE
```

Comment:
```suggestion
        if ($action === 'Capture') { // CAPTURE
```
(there are still about 23 of these left)